ui/app/components/clients/page/counts.hbs

```diff
@@ -118,7 +118,7 @@
         </Hds::Alert>
       {{/if}}
 
-      <Clients::Counts::NavBar @showSecretsSyncClientCounts={{or this.hasSecretsSyncClients this.flags.showSecretsSync}} />
+      <Clients::Counts::NavBar @showSecretsSyncClientCounts={{or this.hasSecretsSyncClients this.version.hasSecretsSync}} />
```

Comment:
Didn't include logic for `hasNavPermission` here because navigating to the sync route is irrelevant to viewing sync client count data (see [comment](https://github.com/hashicorp/vault/pull/30001/files#r2008495393) for more context)